### DISCUSSION TITLE
Sftp and ftp on default ports

### DISCUSSION THREAD

**Discussion by jeanmeuze** [drakkan/sftpgo#1351] [Q&A] [answered]
I installed sftpgo on CentOs and would like to have both sftp and ftp running on the default ports (22 & 21 resp.)
I was able to set ftp on port 21 by stopping the vsftpd and overruling the sftpgo.json property using env

But to get sftp on port 22, I tried adding it as subsystem to sshd
Everything starts up without errors, but I can not log in using sftp on port 22
Journalctl gives this error:
Jul 07 14:21:42 sftpgo-01 sshd[7413]: Invalid user _username_ from _ip-address_ port 56031
Jul 07 14:21:42 sftpgo-01 sshd[7413]: input_userauth_request: invalid user _username_[preauth]
Jul 07 14:21:44 sftpgo-01 sshd[7413]: pam_unix(sshd:auth): check pass; user unknown
Jul 07 14:21:44 sftpgo-01 sshd[7413]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=dell-5fzdyy2.vm.be
Jul 07 14:21:46 sftpgo-01 sshd[7413]: Failed password for invalid user _username_ from  _ip-address_ port 56031 ssh2

So it does seem to find the configured users (defined in the web console) but does not accept the login/password

What am I missing here?
Or is there a better approach to combine ftp on port 21, sftp on port 22 and the web admin to manage it all?

**Comment by ben790924** [comment]
I struggled changing sftp(default 2022) port to 22 as well. 
I could only come up with ENV vars the document menthioned(SFTPGO_SFTPD__BINDINGS__0__PORT) and alter sftpgo.json but not working.

When I change sftpgo.json (sftpd.bindings[0].port) and restart service it will give error.

![image](https://github.com/drakkan/sftpgo/assets/37503754/769b5271-d54b-4b12-8244-fba47122d40d)


Have you solved this issue?

**Reply by jeanmeuze** [reply]
No, I have not found a solution yet on how to use the sftp as subsystem on ssh on port 22.
However, I did manage to get it running by changing the sftpd port to 22 (as you did, but I used the env.d folder) and by changing the sshd port to something else
So I'm considering to use it like this with the sshd running on some other port if I can't get it running as a sshd subsystem

**Reply by ben790924** [reply]
I was silly I used port 22 to access VM resulted in 22 port in used. got my issue solved as well. Thanks

**Comment by drakkan** [comment, marked as answer]
> No, I have not found a solution yet on how to use the sftp as subsystem on ssh on port 22. However, I did manage to get it running by changing the sftpd port to 22 (as you did, but I used the env.d folder) and by changing the sshd port to something else So I'm considering to use it like this with the sshd running on some other port if I can't get it running as a sshd subsystem

Please read this [doc](https://github.com/drakkan/sftpgo/blob/main/docs/howto/openssh-sftp-subsystem.md) to understand the subsystem mode. SFTPGo does not handle authentication in this mode and some features are not available.

**Reply by jeanmeuze** [reply]
Ah! I seem to have missed a small bit of text on that doc:
In this mode **OpenSSH** accepts the network connection, **handles** the SSH handshake and **user authentication**
Once I created the same user as a linux user on the system, I was able to sftp with that username/password on port 22
However, giving the extra complexity to handle the users and the other limitations, I'm still considering setting it up with a dedicated port 22 for sftp and some other port for pure ssh access
Thanks

**Reply by drakkan** [reply]
yes, subsystem mode is for very specific use cases, you should use SFTPGo in standalone mode and change the port if you want it to port 22
